ocrd/ocrd/processor/helpers.py

```diff
@@ -69,7 +71,13 @@ def run_processor(
     log.debug("Processor instance %s (%s doing %s)", processor, name, otherrole)
     t0_wall = perf_counter()
     t0_cpu = process_time()
-    processor.process()
+    mem_usage = memory_usage(proc=processor.process, interval=.1, timeout=None, timestamps=True)
```

Comment:
Maybe you wanna add `multiprocess=True` in there for processors that use `multiprocessing` or similar, or that delegate to shell calls. Perhaps also `include_children=True` (not sure about the meaning though – does the sum then count shared/CoW pages repeatedly?).

Reply:
> > Perhaps also `include_children=True` (not sure about the meaning though – does the sum then count shared/CoW pages repeatedly?).
> 
> ```python
>             if include_children:
>                 mem +=  sum(_get_child_memory(process, meminfo_attr))
> ```

I saw that, but the question remains: what is in that sum? (We probably don't want to count shared and CoW pages repeatedly. But it may be hard to calculate exactly. IIRC standard `ps` and `top` and even `time` do count repetitions, so they appear to large (don't "add up") for multiprocessed programs, whereas `smem` gets it right. [This](https://unix.stackexchange.com/questions/34795/correctly-determining-memory-usage-in-linux) tells me we are actually interested in the PSS, the proportional set size, rather than the RSS.)

Reply:
Ok, but PSS vs RSS is a more general problem/discussion. Any multiprocessing coder will know about that. And RSS is still a valuable information. So let's stick to your current implementation.